---
title: "Adding General and NF-CORE Pipelines"
output:
  html_document:
    toc: true
    toc_float: false
    toc-expand: 2
---

## General pipelines

TBA

### Example: Hello World

A simple script showing the basic Hello World! example for the Nextflow framework.

![https://github.com/nextflow-io/hello](images/hello.png)

![](images/helloworlddiagram.png)

<div>

> -   TBA
>
> -   TBA
>
> -   TBA
>
> -   TBA

</div>

## NF-CORE pipelines

[nf-core](https://nf-co.re/pipelines) is a community effort to collect a curated set of analysis pipelines built using Nextflow. There are 86 pipelines that are currently available.\

### Example: Germline-StructuralV-nf pipeline

![GermlineStructuralV-nf is a pipeline for identifying structural variant events in human Illumina short read whole genome sequence data. GermlineStructuralV-nf identifies structural variant and copy number events from BAM files using Manta, Smoove, and TIDDIT. Variants are then merged using SURVIVOR, and annotated by AnnotSV. The pipeline is written in Nextflow and uses Singularity/Docker to run containerised tools.](images/211971740-772796bc-6fb7-43fb-885b-d9cb116bfdd0.png)

> -   From the Launchpad, select "Add pipeline". 
>
> -   Give your pipeline a descriptive name. It needs to be unique to your community (can't create one that already exists in broader BioCommons group). 
>
> -   Select your previously configured compute environment.
>
> -   Provide pipeline codebase URL <https://github.com/Sydney-Informatics-Hub/Germline-StructuralV-nf>. 
>
> -   Select **main** for the Revision number.
>
> -   Specify \$TW_AGENT_WORK for the work directory.
>
> -   Select either gadi or setonix for the config profiles (these are provided in the code base of the workflow).
>
> -   Select "Add".
>
>     ![](images/Screen%20Shot%202023-08-25%20at%204.35.01%20pm.png)

Your launchpad will now reload with a preconfigured workflow.

![](images/Screen%20Shot%202023-08-25%20at%204.36.57%20pm.png)

\
